crates/client/commitment-state-diff/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+where
+    C: ProvideRuntimeApi<B>,
+    C::Api: StarknetRuntimeApi<B>,
+    C: HeaderBackend<B>,
+    H: HasherT,
+{
+    let starknet_block_hash = {
+        let header = client.header(storage_notification.block)?.ok_or(BuildCommitmentStateDiffError::BlockNotFound)?;
+        let digest = header.digest();
+        let block = mp_digest_log::find_starknet_block(digest)?;
+        block.header().hash::<H>().into()
+    };
+
+    let mut commitment_state_diff = CommitmentStateDiff {
+        address_to_class_hash: Default::default(),
+        address_to_nonce: Default::default(),
+        storage_updates: Default::default(),
+        class_hash_to_compiled_class_hash: Default::default(),
+    };
+
+    for (_prefix, full_storage_key, change) in storage_notification.changes.iter() {
+        // The storages we are interested in here all have longe keys
+        if full_storage_key.0.len() < 32 {
+            continue;
+        }
+        let prefix = &full_storage_key.0[..32];
+
+        // All the try_into are safe to unwrap because we know that is what the storage contains
+        // and therefore what length it is
+        if prefix == *SN_NONCE_PREFIX {
+            let contract_address =
+                ContractAddress(PatriciaKey(StarkFelt(full_storage_key.0[32..].try_into().unwrap())));
+            // `change` is safe to unwrap as `Nonces` storage is `ValueQuery`
+            let nonce = Nonce(StarkFelt(change.unwrap().0.clone().try_into().unwrap()));
+            commitment_state_diff.address_to_nonce.insert(contract_address, nonce);
+        } else if prefix == *SN_STORAGE_PREFIX {
+            let contract_address =
+                ContractAddress(PatriciaKey(StarkFelt(full_storage_key.0[32..64].try_into().unwrap())));
+            let storage_key = StarknetStorageKey(PatriciaKey(StarkFelt(full_storage_key.0[64..].try_into().unwrap())));
+            // `change` is safe to unwrap as `StorageView` storage is `ValueQuery`
+            let value = StarkFelt(change.unwrap().0.clone().try_into().unwrap());
+
+            match commitment_state_diff.storage_updates.get_mut(&contract_address) {
+                Some(contract_storage) => {
+                    contract_storage.insert(storage_key, value);
+                }
+                None => {
+                    let mut contract_storage: IndexMap<_, _, _> = Default::default();
+                    contract_storage.insert(storage_key, value);
+
+                    commitment_state_diff.storage_updates.insert(contract_address, contract_storage);
+                }
+            }
+        } else if prefix == *SN_CONTRACT_CLASS_HASH_PREFIX {
+            let contract_address =
+                ContractAddress(PatriciaKey(StarkFelt(full_storage_key.0[32..].try_into().unwrap())));
+            // `change` is safe to unwrap as `ContractClassHashes` storage is `ValueQuery`
+            let class_hash = ClassHash(StarkFelt(change.unwrap().0.clone().try_into().unwrap()));
```

Comment:
we could declare `change.unwrap().0.clone().try_into().unwrap()` outside of the if/else branches

Reply:
`change.unwrap()`
change is not always safe to unwrap.

It would be `None` if the storage was an `OptionValue` that has been set back to none.

Here I wrote:
`// `change` is safe to unwrap as `ContractClassHashes` storage is `ValueQuery``

later you will find:
```rust
            let compiled_class_hash = CompiledClassHash(match change {
                Some(data) => StarkFelt(data.0.clone().try_into().unwrap()),
                // This should not happen in the current state of starknet protocol, but there have been
                // an erase_contract_class mechanism live on the network during the regenesis migration.
                // Better safe than sorry
                None => StarkFelt::default(),
            });
```
where I match change rather than unwrapping it